lib/xcodeproj/project/project_helper.rb

```diff
@@ -64,7 +64,60 @@ def self.new_target(project, type, name, platform, deployment_target, product_gr
 
         target
       end
+      #-----------------------------------------------------------------------#
+
+      # Duplicates a target and adds it to the project.
+      # @param  [Project] project
+      #         the project from which the target should be duplicated.
+      #
+      # @param  [klass] src_target
+      #         the target or the name of the target to duplicate
+      #
+      #
+      # @param  [String] dst_target_name
+      #         the name of the duplicated target.
+      #
+      #
+      # @return [PBXNativeTarget] the target.
+      #
+      def self.duplicate_target(project, src_target, dst_target_name)
+        if src_target.is_a?(String)
```

Comment:
Check if `src_target.is_a?(PBXNativeTarget)` and use otherwise `src_target.to_s`, otherwise you expect that the parameter has a specific type without checking it.